src/components/profile/verificationModals/confirmVerificationModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button, Form, Modal } from "react-bootstrap";
+import { IVA, IVAStatus } from "../../../models/ivas";
+import { useState } from "react";
+import { AUTH_URL, fetchJson } from "../../../utils/utils";
+import { showMessage } from "../../messages/usage";
+
+interface ConfirmVerificationModalProps {
+  show: boolean;
+  setShow: any;
+  toConfirmIVA: IVA | null;
+  setToConfirmIVA: any;
+}
+
+const ConfirmVerificationModal = (props: ConfirmVerificationModalProps) => {
+  const HandleSubmit = async (verificationCode: string) => {
+    setDisabledButton(true);
+    let url = AUTH_URL;
+    url = new URL(`rpc/ivas/${props.toConfirmIVA!.id}/validate-code`, url);
+    let method: string = "POST",
+      ok: number = 204,
+      userData = { verification_code: verificationCode };
+    const response = await fetchJson(url, method, userData).catch(() => null);
+    if (response && response.status === ok) {
+      props.toConfirmIVA!.status = IVAStatus.Verified;
+      props.setShow(false);
+      props.setToConfirmIVA(null);
+      showMessage({
+        type: "success",
+        title: "Contact address verified successfully",
+      });
+    } else if (response && response.status === 429) {
+      props.toConfirmIVA!.status = IVAStatus.Unverified;
+      showMessage({
+        type: "error",
+        title:
+          "Too many attempts at entering code. Contact address has been reverted to unverified",
+      });
+    } else {
+      showMessage({
+        type: "error",
+        title: "Code could not be verified. Please try again later",
+      });
+      setTimeout(() => setDisabledButton(false), 2000);
+    }
+  };
+
+  const [disabledButton, setDisabledButton] = useState(true);
+  const [inputtedCode, setInputtedCode] = useState("");
+
+  return (
+    <>
+      {props.toConfirmIVA ? (
+        <Modal
+          show={props.show}
+          onHide={() => {
+            props.setShow(false);
+          }}
+        >
+          <Modal.Header closeButton>
+            <Modal.Title>Verification Address Confirmation</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form
+              onSubmit={(e) => {
+                e.preventDefault();
+                HandleSubmit(inputtedCode);
+              }}
+            >
+              <p>
+                We sent a verification code to your selected contact address.
+                Please enter it here:
+              </p>
+              <input
+                type="text"
+                name="verificationCode"
+                id="verificationCode"
+                className="fs-2 ps-4"
+                style={{ letterSpacing: "0.5em", paddingLeft: "0.5em" }}
+                required
+                minLength={6}
+                maxLength={6}
+                pattern="[0-9]{6}"
```

Comment:
```suggestion
                pattern="[A-Z09-9]{6}"
```

Verification codes have 6 uppercase letter or digits, not just digits (and they always have a mix of both), so they cannot be confused with TOTP codes.

Reply:
Done 👍